src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -27,11 +29,13 @@
 class SendFailedMessageToFailureTransportListener implements EventSubscriberInterface
 {
     private ContainerInterface $failureSenders;
+    private ContainerInterface $retryStrategyLocator;
     private ?LoggerInterface $logger;
 
-    public function __construct(ContainerInterface $failureSenders, ?LoggerInterface $logger = null)
+    public function __construct(ContainerInterface $failureSenders, ContainerInterface $retryStrategyLocator, ?LoggerInterface $logger = null)
```

Comment:
the BC break is not needed and should be avoided
the new argument should be optional and should be added last

Reply:
I agree and just changed it. My reasoning was that the listener without a retry strategy reintroduces the issue of instant and infinite retries when consuming from the failure transport without some limits. But its not worth breaking  builds over this.